test/test_adcs_handler_i2c.c

```diff
@@ -0,0 +1,73 @@
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "adcs_handler.h"
+#include "adcs_io.h"
+#include "mock_uart_i2c.h"
+#include "unity.h"
+
+uint8_t TC_ID_array[] = {1,   4,   5,   6,   33,  108, 112,
+                         113, 114, 115, 116, 117, 118, 119};
+
+void setUp(void) {}
+
+void tearDown(void) {}
+
+void test_ADCS_I2C_send(void) {
+    uint8_t tc_ack_frame[4];
+    tc_ack_frame[0] = LAST_TC_ACK_ID;
+    tc_ack_frame[1] = 1;
+    tc_ack_frame[2] = ADCS_OK;
+    tc_ack_frame[3] = 0;
+    
+    i2c_send_Ignore();
+    i2c_receive_ExpectAnyArgs();
+    i2c_receive_ReturnArrayThruPtr_data(tc_ack_frame, 4);
+    i2c_receive_ExpectAnyArgs();
+    i2c_receive_ReturnArrayThruPtr_data(tc_ack_frame, 4);
```

Comment:
Out of curiosity, is there a reason that you have repeated these lines?

Reply:
This is because i2c_receive() is called twice in the send_i2c_telecommand() function. First time to check the processed flag and then again to check the Error flag.